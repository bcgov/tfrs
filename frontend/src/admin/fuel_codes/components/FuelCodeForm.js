/*
 * Presentational component
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';

import FuelCodeFormDetails from './FuelCodeFormDetails';
import history from '../../../app/History';
import * as Lang from '../../../constants/langEnUs';
import Errors from '../../../app/components/Errors';
import TooltipWhenDisabled from '../../../app/components/TooltipWhenDisabled';

class FuelCodeForm extends Component {
  _getValidationMessagesForDraft () {
    const validationMessage = [];

    if (this.props.fields.fuelCode === '') {
      validationMessage.push('Please enter a fuel code.');
    }

    if (this.props.fields.company === '') {
      validationMessage.push('Please enter a company.');
    }

    if (this.props.fields.carbonIntensity === '') {
      validationMessage.push('Please enter the carbon intensity.');
    }

    if (this.props.fields.fuel === '') {
      validationMessage.push('Please select a fuel.');
    }

    if (this.props.fields.feedstock === '') {
      validationMessage.push('Please enter a feedstock.');
    }

    if (this.props.fields.feedstockLocation === '') {
      validationMessage.push('Please enter a feedstock location.');
    }

    if (this.props.fields.facilityLocation === '') {
      validationMessage.push('Please enter a fuel production facility location.');
    }

    if (this.props.fields.feedstockTransportMode === '') {
      validationMessage.push('Please select a feedstock transport mode.');
    }

    if (this.props.fields.fuelTransportMode === '') {
      validationMessage.push('Please select a finished fuel transport mode.');
    }

    return validationMessage;
  }

  _getValidationMessagesForApproval () {
    const validationMessage = this._getValidationMessagesForDraft();

    if (this.props.fields.applicationDate === '') {
      validationMessage.push('Please enter an application date.');
    }

    if (this.props.fields.effectiveDate === '') {
      validationMessage.push('Please enter an effective date.');
    }

    if (this.props.fields.expiryDate === '') {
      validationMessage.push('Please enter an expiry date.');
    }

    if (this.props.fields.approvalDate === '') {
      validationMessage.push('Please enter a approval date.');
    }

    return validationMessage;
  }

  render () {
    return (
      <div className="page-admin-fuel-code">
        <h1>{this.props.title}</h1>
        <form
          onSubmit={event => this.props.handleSubmit(event)}
        >
          <FuelCodeFormDetails
            addToFields={this.props.addToFields}
            fields={this.props.fields}
            approvedFuels={this.props.approvedFuels}
            transportModes={this.props.transportModes}
            handleInputChange={this.props.handleInputChange}
          />

          {Object.keys(this.props.errors).length > 0 &&
          <Errors errors={this.props.errors} />
          }

          <div className="fuel-code-actions">
            <div className="btn-container">
              <button
                className="btn btn-default"
                onClick={() => history.goBack()}
                type="button"
              >
                <FontAwesomeIcon icon="arrow-circle-left" /> {Lang.BTN_APP_CANCEL}
              </button>
              <TooltipWhenDisabled
                disabled={this._getValidationMessagesForDraft().length > 0}
                title={this._getValidationMessagesForDraft()}
              >
                <button
                  className="btn btn-default"
                  disabled={this._getValidationMessagesForDraft().length > 0}
                  type="submit"
                >
                  <FontAwesomeIcon icon="save" /> {Lang.BTN_SAVE_DRAFT}
                </button>
              </TooltipWhenDisabled>
              <TooltipWhenDisabled
                className="adjust-tooltip"
                disabled={this._getValidationMessagesForApproval().length > 0}
                title={this._getValidationMessagesForApproval()}
              >
                <button
                  className="btn btn-primary"
                  data-target="#confirmSubmit"
                  data-toggle="modal"
                  disabled={this._getValidationMessagesForApproval().length > 0}
                  type="button"
                >
                  <FontAwesomeIcon icon={this.props.edit ? 'save' : 'plus'} />
                  {this.props.edit ? ` ${Lang.BTN_UPDATE}` : ` ${Lang.BTN_ADD}`}
                </button>
              </TooltipWhenDisabled>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

FuelCodeForm.defaultProps = {
  edit: false,
  errors: []
};

FuelCodeForm.propTypes = {
  addToFields: PropTypes.func.isRequired,
  edit: PropTypes.bool,
  errors: PropTypes.shape(),
  fields: PropTypes.shape({
    applicationDate: PropTypes.string,
    approvalDate: PropTypes.string,
    carbonIntensity: PropTypes.string,
    company: PropTypes.string,
    effectiveDate: PropTypes.string,
    expiryDate: PropTypes.string,
    facilityLocation: PropTypes.string,
    facilityNameplate: PropTypes.oneOfType([
      PropTypes.number,
      PropTypes.string
    ]),
    feedstock: PropTypes.string,
    feedstockLocation: PropTypes.string,
    feedstockMisc: PropTypes.string,
    feedstockTransportMode: PropTypes.arrayOf(PropTypes.string),
    formerCompany: PropTypes.string,
    fuel: PropTypes.string,
    fuelCode: PropTypes.string,
    fuelTransportMode: PropTypes.arrayOf(PropTypes.string)
  }).isRequired,
  handleInputChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  approvedFuels: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  transportModes: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  title: PropTypes.string.isRequired
};

export default FuelCodeForm;
