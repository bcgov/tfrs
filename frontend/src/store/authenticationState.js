import { loadUser } from 'redux-oidc';
import { put, takeLatest, all } from 'redux-saga/effects';
import userManager from './oidc-usermanager';
import { getLoggedInUser } from '../actions/userActions';
import CONFIG from '../config';

const LOGIN_TRIGGERING_ACTIONS = [
  'redux-oidc/USER_EXPIRED'
];

function triggerLoginFlow (store) {
  const { routing } = store.getState();

  if (routing.location &&
    routing.location.pathname !== '/authCallback') {
    return CONFIG.KEYCLOAK.CUSTOM_LOGIN ? userManager.signinSilent() : userManager.signinRedirect();
  }

  return false;
}

function * getBackendUser () {
  yield put(getLoggedInUser());
}

export default function * authenticationStateSaga (store) {
  userManager.clearStaleState();

  const { routing } = store.getState();

  if (!routing.location || routing.location.pathname !== '/authCallback') {
    loadUser(store, userManager);
  }
  yield all([
    takeLatest('redux-oidc/USER_FOUND', getBackendUser),
    takeLatest(
      action => (LOGIN_TRIGGERING_ACTIONS.includes(action.type)),
      triggerLoginFlow,
      store
    )
  ]);
}
