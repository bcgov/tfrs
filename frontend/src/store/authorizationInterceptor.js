import axios from 'axios';
import userManager from './oidc-usermanager';
import CONFIG from '../config';

function configureAxios () {
  if (CONFIG.KEYCLOAK) {
    console.log('adding token to request');

    axios.interceptors.request.use((config) => {
      const loadconfig = async () => {
        const user = await userManager.getUser();
        return {
          ...config,
          headers: {
            Authorization: `Bearer ${user.id_token}`
          }
        };
      };

      return loadconfig();
    });
  }
}

export default configureAxios;
