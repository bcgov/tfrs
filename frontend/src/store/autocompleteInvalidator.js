import { call, put, takeLatest } from 'redux-saga/effects';
import AutocompleteActionTypes from "../constants/actionTypes/Autocomplete";
import FuelCodeActionTypes from '../constants/actionTypes/FuelCodes';

const INVALIDATING_ACTIONS = [
  FuelCodeActionTypes.ADD_FUEL_CODE,
  FuelCodeActionTypes.RECEIVE_FUEL_CODE,
  FuelCodeActionTypes.RECEIVE_FUEL_CODES,
  FuelCodeActionTypes.UPDATE_FUEL_CODE,
  FuelCodeActionTypes.DELETE_FUEL_CODE
];

function * invalidateCache () {
  yield put({ type: AutocompleteActionTypes.INVALIDATE_AUTOCOMPLETE_CACHE });
}

export default function * sessionTimeoutSaga () {
  yield takeLatest(
    action => (INVALIDATING_ACTIONS.includes(action.type)),
    invalidateCache
  );
}
