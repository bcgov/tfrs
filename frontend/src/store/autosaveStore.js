import {takeEvery, all, call} from 'redux-saga/effects';
import {delay} from "redux-saga";
import axios from "axios";
import * as Routes from "../constants/routes";

function* loadAutosaveData(action) {
  const {key, resolve, reject} = action.payload;

  axios
    .get(`${Routes.BASE_URL}${Routes.AUTOSAVE_API}?key=${key}`)
    .then((response) => {
      try {
        const parsed = JSON.parse(response.data.data);
        resolve(parsed);
      } catch (e) {
        reject(e);
      }
    }).catch((error) => {
    reject(error);
  });
}

function* saveAutosaveData(action) {
  const {key, state, resolve, reject} = action.payload;

  const payload = {
    key,
    data: JSON.stringify(state)
  };

  axios
    .post(`${Routes.BASE_URL}${Routes.AUTOSAVE_API}`, payload)
    .then((response) => {
      resolve();
    }).catch((error) => {
    reject(error);
  });
}

function* clearAutosaveData(action) {
  const {resolve, reject} = action.payload;

  axios
    .post(`${Routes.BASE_URL}${Routes.AUTOSAVE_API}/clear`)
    .then((response) => {
      try {
        resolve();
      } catch (e) {
        reject(e);
      }
    }).catch((error) => {
    reject(error);
  });
}

export default function* autosaveSaga() {

  yield all([
    takeEvery('LOAD_AUTOSAVE_DATA', loadAutosaveData),
    takeEvery('SAVE_AUTOSAVE_DATA', saveAutosaveData),
    takeEvery('CLEAR_AUTOSAVE_DATA', clearAutosaveData)
  ]);
}
