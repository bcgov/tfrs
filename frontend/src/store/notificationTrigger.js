import {call, put, takeLatest} from 'redux-saga/effects';
import {delay} from 'redux-saga';
import {getNotifications, getNotificationsCount} from "../actions/notificationActions";
import UserActionTypes from "../constants/actionTypes/Users";
import NotificationActionTypes from "../constants/actionTypes/Notifications";


const TRIGGERING_ACTIONS = [
  'SERVER_INITIATED_NOTIFICATION_RELOAD',
  UserActionTypes.RECEIVE_LOGGED_IN_USER,
  NotificationActionTypes.SUCCESS_NOTIFICATIONS
];

function* fetchNotifications(store) {
  yield call(delay, 1000); //debounce

  if (store.getState().rootReducer.userRequest.isAuthenticated) {
    yield put(getNotificationsCount());
  }

  if (store.getState().rootReducer.notifications.onNotificationsPage) {
    yield put(getNotifications());
  }

}

export default function* notificationsSaga(store) {
  yield takeLatest(
    action => (TRIGGERING_ACTIONS.includes(action.type)),
    fetchNotifications,
    store
  );
}
