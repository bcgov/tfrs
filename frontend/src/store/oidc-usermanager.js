import { createUserManager } from 'redux-oidc';
import CONFIG from '../config';

const settings = {
  authority: CONFIG.KEYCLOAK.AUTHORITY,
  client_id: CONFIG.KEYCLOAK.CLIENT_ID,
  redirect_uri: CONFIG.KEYCLOAK.CALLBACK_URL,
  automaticSilentRenew: true,
  silent_redirect_uri: `${window.location.protocol}//${window.location.hostname}${window.location.port ? `:${window.location.port}` : ''}/token_renew.html`,
  response_type: 'token id_token'
};

const userManager = createUserManager(settings);

export default userManager;
