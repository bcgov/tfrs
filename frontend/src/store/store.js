import { createStore, compose, applyMiddleware, combineReducers } from 'redux';
import { reducer as toastrReducer } from 'react-redux-toastr';
import { routerReducer, routerMiddleware } from 'react-router-redux';
import { createLogger } from 'redux-logger';
import io from 'socket.io-client';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/reducer';
import { getNotifications } from '../actions/notificationActions';
import { SOCKETIO_URL } from '../constants/routes';

import createOidcMiddleware, { loadUser, processSilentRenew, reducer as OIDCReducer } from 'redux-oidc';
import { persistTargetPathReducer } from '../reducers/persistTargetPathReducer';

import userManager from './oidc-usermanager';
import CONFIG from '../config';
import { getLoggedInUser } from '../actions/userActions';

import persistState from 'redux-localstorage';
import createSocketIoMiddleware from 'redux-socket.io';

const middleware = routerMiddleware(history);
const oidcMiddleware = createOidcMiddleware(userManager);

const socket = io(SOCKETIO_URL);
const socketIoMiddleware = createSocketIoMiddleware(socket, 'socketio/');

const enhancer = compose(persistState(['targetPath'], {key: 'tfrs-state'}));

const combinedReducers = (state = {}, action) => {
  const currentRoute = state.routing || {};
  return {
    toastr: toastrReducer(state.toastr, action),
    oidc: OIDCReducer(state.oidc, action),
    routing: routerReducer(state.routing, action),
    targetPath: persistTargetPathReducer(state.targetPath, {...action, currentRoute}),
    rootReducer: rootReducer(state.rootReducer, action),
  };
};

let allMiddleware = [
  thunk,
  socketIoMiddleware,
  oidcMiddleware,
  middleware
];

if (process.env.NODE_ENV !== 'production') {
  allMiddleware.push(createLogger());
}

const store = createStore(
  combinedReducers,
  applyMiddleware(...allMiddleware),
  enhancer
);

let subscriptionProcessing = false;

store.subscribe(() => {
  const state = store.getState();
  if (!subscriptionProcessing) {
    subscriptionProcessing = true;

    if (state.rootReducer.notifications.serverInitiatedReloadRequested === true) {
      store.dispatch(getNotifications());
    }

    if (CONFIG.KEYCLOAK.ENABLED) {
      if (state.oidc.user &&
        !state.rootReducer.userRequest.isFetching &&
        !state.rootReducer.userRequest.isAuthenticated) {
        store.dispatch(getLoggedInUser());
      }
    }

    subscriptionProcessing = false;
  }
});

if (CONFIG.KEYCLOAK.ENABLED) {
  loadUser(store, userManager);
  processSilentRenew();
}

export default store;
