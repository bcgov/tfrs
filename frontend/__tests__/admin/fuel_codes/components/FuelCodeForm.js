import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import FuelCodeForm from '../../../../src/admin/fuel_codes/components/FuelCodeForm';
import store from '../../../../src/store/store';

describe('Test Fuel Code Form', () => {
  const approvedFuels = [{
    name: 'LNG'
  }, {
    name: 'Propane'
  }];
  const errors = {};
  const fields = {
    feedstockTransportMode: [],
    fuelTransportMode: []
  };
  const transportModes = [{
    name: 'Truck'
  }, {
    name: 'Rail'
  }];

  const component = renderer.create(
    <Provider store={store}>
      <FuelCodeForm
        addToFields={() => {}}
        approvedFuels={approvedFuels}
        errors={errors}
        fields={fields}
        handleInputChange={() => {}}
        handleSubmit={() => {}}
        title="Edit Fuel"
        transportModes={transportModes}
      />
    </Provider>
  );

  test('FuelCodeForm should display', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
