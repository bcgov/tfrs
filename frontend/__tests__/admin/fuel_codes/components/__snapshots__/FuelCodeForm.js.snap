// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Fuel Code Form FuelCodeForm should display 1`] = `
<div
  className="page-admin-fuel-code"
>
  <h1>
    Edit Fuel
  </h1>
  <form
    onSubmit={[Function]}
  >
    <div
      className="fuel-code-details"
    >
      <div
        className="main-form"
      >
        <div
          className="row"
        >
          <div
            className="col-sm-6"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="fuel-code"
              >
                Low Carbon Fuel Code:
                <div
                  className="input-group"
                >
                  <span
                    className="input-group-addon"
                  >
                    BCLCF
                  </span>
                  <input
                    className="form-control"
                    id="fuel-code"
                    name="fuelCode"
                    onChange={[Function]}
                    required="required"
                    type="text"
                  />
                </div>
              </label>
            </div>
          </div>
          <div
            className="col-sm-6"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="company"
              >
                Company:
                <div
                  style={Object {}}
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded={false}
                    autoComplete="off"
                    className="form-control"
                    id="company"
                    name="company"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    required={true}
                    role="combobox"
                    type="text"
                    value=""
                  />
                </div>
              </label>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-6"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="carbon-intensity"
              >
                Carbon Intensity:
                <input
                  className="form-control"
                  id="carbon-intensity"
                  name="carbonIntensity"
                  onChange={[Function]}
                  required="required"
                  step="0.01"
                  type="number"
                />
              </label>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-6"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="application-date"
              >
                Application Date:
                <input
                  className="form-control"
                  id="application-date"
                  max="9999-12-31"
                  name="applicationDate"
                  onChange={[Function]}
                  type="date"
                />
              </label>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-6"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="effective-date"
              >
                Effective Date:
                <input
                  className="form-control"
                  id="effective-date"
                  max="9999-12-31"
                  name="effectiveDate"
                  onChange={[Function]}
                  type="date"
                />
              </label>
            </div>
          </div>
          <div
            className="col-sm-6"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="expiry-date"
              >
                Expiry Date:
                <input
                  className="form-control"
                  id="expiry-date"
                  max="9999-12-31"
                  name="expiryDate"
                  onChange={[Function]}
                  type="date"
                />
              </label>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-6"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="fuel"
              >
                Fuel:
                <select
                  className="form-control"
                  id="fuel"
                  name="fuel"
                  onChange={[Function]}
                  required="required"
                >
                  <option
                    default={true}
                    value=""
                  />
                  <option
                    value="LNG"
                  >
                    LNG
                  </option>
                  <option
                    value="Propane"
                  >
                    Propane
                  </option>
                </select>
              </label>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-6"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="feedstock"
              >
                Feedstock:
                <div
                  style={Object {}}
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded={false}
                    autoComplete="off"
                    className="form-control"
                    id="feedstock"
                    name="feedstock"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    required={true}
                    role="combobox"
                    type="text"
                    value=""
                  />
                </div>
              </label>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-6"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="feedstock-location"
              >
                Feedstock Location:
                <div
                  style={Object {}}
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded={false}
                    autoComplete="off"
                    className="form-control"
                    id="feedstockLocation"
                    name="feedstockLocation"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    required={true}
                    role="combobox"
                    type="text"
                    value=""
                  />
                </div>
              </label>
            </div>
          </div>
          <div
            className="col-sm-6"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="feedstock-miscellaneous"
              >
                Feedstock Misc:
                <div
                  style={Object {}}
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded={false}
                    autoComplete="off"
                    className="form-control"
                    id="feedstockMisc"
                    name="feedstockMisc"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    required={false}
                    role="combobox"
                    type="text"
                    value=""
                  />
                </div>
              </label>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-6"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="facility-location"
              >
                Fuel Production Facility Location (City, Province/State/Country):
                <div
                  style={Object {}}
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded={false}
                    autoComplete="off"
                    className="form-control"
                    id="facilityLocation"
                    name="facilityLocation"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    required={true}
                    role="combobox"
                    type="text"
                    value=""
                  />
                </div>
              </label>
            </div>
          </div>
          <div
            className="col-sm-6"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="facility-nameplate"
              >
                Fuel Production Facility Nameplate Capacity (litres/GJ per year):
                <input
                  className="form-control"
                  id="facility-nameplate"
                  name="facilityNameplate"
                  onChange={[Function]}
                  type="number"
                />
              </label>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-6"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="feedstock-transport-mode"
              >
                Feedstock Transport Mode (use ctrl key to select multiple modes):
                <select
                  className="form-control"
                  id="feedstock-transport-mode"
                  multiple={true}
                  name="feedstockTransportMode"
                  onChange={[Function]}
                >
                  <option
                    value="Truck"
                  >
                    Truck
                  </option>
                  <option
                    value="Rail"
                  >
                    Rail
                  </option>
                </select>
              </label>
            </div>
          </div>
          <div
            className="col-sm-6"
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="fuel-transport-mode"
              >
                Finished Fuel Transport Mode (use ctrl key to select multiple modes):
                <select
                  className="form-control"
                  id="fuel-transport-mode"
                  multiple={true}
                  name="fuelTransportMode"
                  onChange={[Function]}
                >
                  <option
                    value="Truck"
                  >
                    Truck
                  </option>
                  <option
                    value="Rail"
                  >
                    Rail
                  </option>
                </select>
              </label>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-6"
          >
            <label
              htmlFor="former-company-name"
            >
              Former Company Name:
              <div
                style={Object {}}
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded={false}
                  autoComplete="off"
                  className="form-control"
                  id="formerCompany"
                  name="formerCompany"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  required={false}
                  role="combobox"
                  type="text"
                  value=""
                />
              </div>
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-sm-6"
          >
            <label
              htmlFor="approval-date"
            >
              Approval Date:
              <input
                className="form-control"
                id="approval-date"
                max="9999-12-31"
                name="approvalDate"
                onChange={[Function]}
                type="date"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      className="fuel-code-actions"
    >
      <div
        className="btn-container"
      >
        <button
          className="btn btn-default"
          onClick={[Function]}
          type="button"
        >
           
          Back
        </button>
        <button
          className="btn btn-default"
          disabled={false}
          type="submit"
        >
           
          Save Draft
        </button>
        <button
          className="btn btn-primary"
          data-target="#confirmSubmit"
          data-toggle="modal"
          disabled={false}
          type="button"
        >
           Add
        </button>
      </div>
    </div>
  </form>
</div>
`;
